5_data_munge/src/munge_data_files.R

```diff
@@ -40,74 +40,144 @@ munge_ecosheds <- function(in_ind, min_value, max_value, out_ind) {
            mean < max_value)
 
   message(nrow(dat) - nrow(dat_out), ' temperature observations dropped because they were outside of 0-35 deg C.')
-  
+
   dat_out2 <- dat_out %>%
     filter(!grepl('delete data|inaccurate|setup|bad value|preplacement|dewatered|out of water|out ofwater|outofwater|removal|set up|too warm', comment, ignore.case = TRUE))
-  
+
   message(nrow(dat_out) - nrow(dat_out2), ' temperature observations dropped due to comments that indicated poor data quality.')
-  
-  saveRDS(dat_out2, file = as_data_file(out_ind))  
+
+  saveRDS(dat_out2, file = as_data_file(out_ind))
+  gd_put(out_ind)
+
+}
+
+munge_norwest <- function(in_ind, min_value, max_value, out_ind) {
+  files <- yaml::read_yaml(in_ind)
+  in_ind <- as_ind_file(names(files)[1])
+  sites_ind <- as_ind_file(names(files)[2])
+  dat <- feather::read_feather(sc_retrieve(in_ind, 'getters.yml'))
+  sites <- readRDS(sc_retrieve(sites_ind, 'getters.yml')) %>%
+    sf::st_drop_geometry() %>%
+    mutate(site_meta = TRUE) %>%
+    select(OBSPRED_ID, Source, UOM, region, site_meta) %>% distinct()
+  dat_out <-  dat %>%
+    mutate(NorWeST_ID = ifelse(is.na(NorWeST_ID), NoRWeST_ID, NorWeST_ID),
+           DailySD = ifelse(is.na(DailySD), DAILYSD, DailySD)) %>%
+    mutate(NorWeST_ID = ifelse(is.na(NorWeST_ID), NorWest_ID, NorWeST_ID)) %>%
+    select(-NoRWeST_ID, -NorWest_ID, -DAILYSD) %>%
+    left_join(sites, by = c('region', 'OBSPRED_ID')) %>%
+    filter(!grepl('nwis', Source, ignore.case = TRUE)) %>% # assume we picked these up in our own NWIS call
+    filter(!is.na(DailyMean)) %>%
+    filter(!is.na(site_meta)) # there were 32 sites from the "OregonCoast" files that did not have metadata
+
+  # handle units
+  # most have units, but 2.4 million do not
```

Comment:
:open_mouth:

Reply:
I expected these data to be really polished....and they were not.

Reply:
yeah, bummer.